negotiation/neural/model_builder.py

```diff
@@ -226,12 +242,26 @@ def make_base_model(model_opt, mappings, gpu, checkpoint=None):
                 p.data.uniform_(-model_opt.param_init, model_opt.param_init)
             for p in generator.parameters():
                 p.data.uniform_(-model_opt.param_init, model_opt.param_init)
+
+        if isinstance(model_opt.pretrained_wordvec, list):
+          dialogue_wordvec = model_opt.pretrained_wordvec[0]
+          kb_wordvec = model_opt.pretrained_wordvec[1]
+        else:
+          dialogue_wordvec = model_opt.pretrained_wordvec
+
         if hasattr(model.encoder, 'embeddings'):
-            model.encoder.embeddings.load_pretrained_vectors(
-                    model_opt.pretrained_wordvec, model_opt.fix_pretrained_wordvec)
+            if model.encoder.embed_type == 'dialogue':
+              model.encoder.embeddings.load_pretrained_vectors(
+                    dialogue_wordvec, model_opt.fix_pretrained_wordvec)
+            elif model.encoder.embed_type == 'kb':
+              model.encoder.embeddings.load_pretrained_vectors(
+                    kb_wordvec, model_opt.fix_pretrained_wordvec)
+            elif model.encoder.embed_type == 'category':
+              model.encoder.embeddings.load_pretrained_vectors(
+                    cat_wordvec, model_opt.fix_pretrained_wordvec)
```

Comment:
we don't have category vocal

Reply:
also encoder only uses dialogue_vocab right?

Reply:
removed the "category" vocab, i'm seeing that your code style is to not write out snippets until we need them.
The encoder and context_embedder both use "dialogue_wordvec", and only "dialogue_wordvec".
The kb_embedder uses "kb_wordvec" and only "kb_wordvec".

The way it works now is each of the 3 embedding models have a "embed_type" attribute signifying the (one) vocab type to load into self.embeddings.  that embed_type is either "dialogue" or "kb".  Thus, we are already loading the pytorch embeddings for all three.